hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -452,6 +452,55 @@ object EmitStream {
           }
         }
 
+
+      case SeqSample(totalSize, numToSample, _requiresMemoryManagementPerElement) =>
+        // Implemented based on http://proceedings.mlr.press/v130/shekelyan21a/shekelyan21a.pdf
+        emit(totalSize, cb).flatMap(cb) { totalSizeCode =>
+          emit(numToSample, cb).map(cb) { numToSampleCode =>
+            val totalSizeVal = totalSizeCode.asInt.memoize(cb, "seq_sample_total_size")
+            val numToSampleVal = numToSampleCode.asInt.memoize(cb, "seq_sample_num_to_sample")
+
+            val len = mb.genFieldThisRef[Int]("seq_sample_len")
+            val regionVar = mb.genFieldThisRef[Region]("seq_sample_region")
+
+            val nRemaining = cb.newLocal[Int]("seq_sample_num_remaining", numToSampleVal.intCode(cb))
+            val candidate = cb.newLocal[Int]("seq_sample_candidate", 0)
+            val elementToReturn = cb.newLocal[Int]("seq_sample_element_to_return", -1) // -1 should never be returned.
+
+            val producer = new StreamProducer {
+              override val length: Option[EmitCodeBuilder => Code[Int]] = Some(_ => len)
+
+              override def initialize(cb: EmitCodeBuilder): Unit = {
+                cb.assign(len, numToSampleVal.asInt.intCode(cb))
```

Comment:
I think you need to initialize all your state variables here (nRemaining, candidate, elementToReturn)